spark/src/main/scala/geotrellis/spark/pyramid/Pyramid.scala

```diff
@@ -42,6 +65,69 @@ object Pyramid extends LazyLogging {
     implicit def methodToOptions(m: ResampleMethod): Options = Options(resampleMethod = m)
   }
 
+  def read[
+    K: AvroRecordCodec: Boundable: JsonFormat: ClassTag: SpatialComponent,
+    V <: CellGrid: ? => TilePrototypeMethods[V]: ? => TileMergeMethods[V]: AvroRecordCodec: ClassTag,
+    M: JsonFormat: Component[?, Bounds[K]]: Component[?, LayoutDefinition]
+  ](layerName: String, maxZoom: Int, minZoom: Int, layerReader: LayerReader[LayerId]): Pyramid[K, V, M] = {
+    val seq = for (z <- maxZoom to minZoom by -1) yield {
+      (z, layerReader.read[K, V, M](LayerId(layerName, z)))
+    }
+    Pyramid[K, V, M](seq.toMap)
+  }
+
+  def write[
+    K: AvroRecordCodec: Boundable: JsonFormat: ClassTag: SpatialComponent,
+    V <: CellGrid: ? => TilePrototypeMethods[V]: ? => TileMergeMethods[V]: AvroRecordCodec: ClassTag,
+    M: JsonFormat: Component[?, Bounds[K]]: Component[?, LayoutDefinition]
+  ](pyramid: Pyramid[K, V, M], layerName: String, writer: LayerWriter[LayerId], keyIndexMethod: KeyIndexMethod[K]) = {
+    for (z <- pyramid.maxZoom to pyramid.minZoom by -1) {
+      writer.write[K, V, M](LayerId(layerName, z), pyramid(z), keyIndexMethod)
+    }
+  }
+
+  def apply[
+    K: SpatialComponent: ClassTag,
+    V <: CellGrid: ClassTag: ? => TilePrototypeMethods[V]: ? => TileMergeMethods[V],
+    M: Component[?, LayoutDefinition]: Component[?, Bounds[K]]
+  ](rdd: RDD[(K, V)] with Metadata[M],
+    startZoom: Int,
+    endZoom: Int,
+    options: Options
```

Comment:
I would suggest unrolling `Options` into `ResampleMethod` and `Partitioner`.
Having the `Options` object solves the problem of bringing a lot of configurations from   boring to spark context but in spark if we're only adding `Partitioner` its more clear to have that as its own option.

I've followed this intuition in rasterize API for reference. What do you think of that approach?